TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/RunningActivity.java

```diff
@@ -341,8 +341,10 @@ protected void init(NodeMainExecutor nodeMainExecutor) {
                         != TangoInitializationHelper.ARCH_ERROR) {
             mTangoRosNode = new TangoRosNode();
             mTangoRosNode.attachCallbackListener(this);
+
+            TangoInitializationHelper.checkTangoVersionOk(this);
             TangoInitializationHelper.bindTangoService(this, mTangoServiceConnection);
-            if (TangoInitializationHelper.checkTangoVersionOk(this)) {
+            if (TangoInitializationHelper.isTangoVersionOk()) {
```

Comment:
I think the implementation of `checkTangoVersionOk` is confusing. It is not clear that it needs to be called for the `mTangoVersionOk` variable to be set. Can you make this a private function, rename to `testTangoVersionOk` and call it inside bindTangoService?